src/cp/crystaldamage.cxx

```diff
@@ -0,0 +1,631 @@
+{
+  auto names = total.items();
+  for (auto name : varnames_) {
+    names.erase(std::remove(names.begin(), names.end(), name), names.end());
+  }
+
+  return total.subset(names);
+}
+
+WorkPlaneDamage::WorkPlaneDamage() :
+    SlipPlaneDamage()
+{
+
+}
+
+std::string WorkPlaneDamage::type()
+{
+  return "WorkPlaneDamage";
+}
+
+std::unique_ptr<NEMLObject> WorkPlaneDamage::initialize(ParameterSet & params)
+{
+  return neml::make_unique<WorkPlaneDamage>();
+}
+
+ParameterSet WorkPlaneDamage::parameters()
+{
+  ParameterSet pset(WorkPlaneDamage::type());
+
+  return pset;
+}
+
+double WorkPlaneDamage::setup() const
+{
+  return 0;
+}
+
+double WorkPlaneDamage::damage_rate(
+    const std::vector<double> & shears, const std::vector<double> & sliprates,
+    double normal_stress, double damage)
+{
+  double rate = 0;
+  for (size_t i = 0; i < shears.size(); i++) 
+    rate += shears[i] * sliprates[i];
+
+  return rate;
+}
+
+std::vector<double> WorkPlaneDamage::d_damage_rate_d_shear(
+    const std::vector<double> & shears, const std::vector<double> & sliprates,
+    double normal_stress, double damage)
+{
+  std::vector<double> deriv;
+  deriv.resize(shears.size());
+  for (size_t i = 0; i < shears.size(); i++)
+    deriv[i] = sliprates[i];
+  return deriv;
+}
+
+std::vector<double> WorkPlaneDamage::d_damage_rate_d_slip(
+    const std::vector<double> & shears, const std::vector<double> & sliprates,
+    double normal_stress, double damage)
+{
+  std::vector<double> deriv;
+  deriv.resize(shears.size());
+  for (size_t i = 0; i < shears.size(); i++)
+    deriv[i] = shears[i];
+  return deriv;
+}
+
+double WorkPlaneDamage::d_damage_rate_d_normal(
+    const std::vector<double> & shears, const std::vector<double> & sliprates,
+    double normal_stress, double damage)
+{
+  return 0;
+}
+
+double WorkPlaneDamage::d_damage_rate_d_damage(
+    const std::vector<double> & shears, const std::vector<double> & sliprates,
+    double normal_stress, double damage)
+{
+  return 0;
+}
+
+SigmoidTransformation::SigmoidTransformation(double c, double beta, double cut) :
+    c_(c), beta_(beta), cut_(cut)
+{
+
+}
+
+std::string SigmoidTransformation::type()
+{
+  return "SigmoidTransformation";
+}
+
+std::unique_ptr<NEMLObject> SigmoidTransformation::initialize(
+    ParameterSet & params)
+{
+  return neml::make_unique<SigmoidTransformation>(
+      params.get_parameter<double>("c"),
+      params.get_parameter<double>("beta"),
+      params.get_parameter<double>("cut"));
+}
+
+ParameterSet SigmoidTransformation::parameters()
+{
+  ParameterSet pset(SigmoidTransformation::type());
+  
+  pset.add_parameter<double>("c");
+  pset.add_parameter<double>("beta");
+  pset.add_optional_parameter<double>("cut", 1.0);
+
+  return pset;
+}
+
+double SigmoidTransformation::map(double damage, double normal_stress)
+{
+  if (damage < 0)
+    return 0;
+  else if (damage < c_)
+  {
+    double dtrial = 1.0/(1.0 + std::pow(damage / (c_ - damage), -beta_));
```

Comment:
This is fine, but can be written more simply as 
`double dtrial = std::pow((c_ - damage) / damage ,beta_)`

Reply:
I can get to `1.0 / (1.0 + std::pow(c_ / damage -1, beta_);` but not all the way to what you have.  I made at least that simplification.